server/schemas/field.graphql

```diff
@@ -0,0 +1,105 @@
+enum FiledType {
+  Text
+  TextArea
+  RichText
+  MarkDownText
+  Asset
+  Date
+  Bool
+  Select
+  Tag
+  Integer
+  Reference
+  URL
+}
+
+type FieldSettings {
+  isMultiValue: Boolean
+  isDeleted: Boolean
+  DefaultValue: Any
+}
+
+type FieldConstraints{
+  isUnique: Boolean
+  isRequired: Boolean
+  #  max: Int
+  #  min: Int
+}
+
+interface IField {
+  id: ID!
+  modelId: ID!
+  model: Model!
+  type: FiledType!
+  key: String!
+  title: String!
+  description: String
+  settings: FieldSettings!
+  constraints: FieldConstraints!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+}
```

Comment:
- Naming of `I~` is common in C#, but not common in Go, JavaScript, and GraphQL. Also this naming is due to the inappropriateness of the name "Field". I suggest "SchemaField" to distinguish it from ordinary Field (it's on Items)
- I think it is better to use interface or union type not for SchemaField, but for SchemaFieldType, since there are no common fields between each types, and if SchemaField was to be an interface, it would contain more fields than necessary and would be cumbersome to implement.

```gql
type SchemaField {
  type SchemaFieldType
}

union SchemaFieldType = SchemaFIeldTypeText | SchemaFieldTypeTextarea

type SchemaFIeldTypeText {

}

type SchemaFieldTypeTextarea {

}
```